src/controllers/profile.controller.ts

```diff
@@ -0,0 +1,71 @@
+import { type Request, type Response } from 'express'
+import { dataSource } from '../configs/dbConfig'
+import Profile from '../entities/profile.entity'
+
+export const getProfile = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    res.status(200).json(req.user)
+  } catch (err) {
+    console.error('Error executing query', err)
+    res.status(500).json({ error: err })
+  }
+}
+
+export const updateProfile = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    const user: Profile | undefined = req.user as Profile
+
+    const {
+      primary_email,
+      contact_email,
+      first_name,
+      last_name,
+      image_url,
+      linkedin_url
+    } = req.body
+
+    const profileRepository = dataSource.getRepository(Profile)
+
+    const updatedProfile = await profileRepository
```

Comment:
I think this has to be done in the service layer. (query building and validations)

Reply:
yeah I separated it.